core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -227,6 +222,11 @@ private[spark] class Executor(
 
   metricsPoller.start()
 
+  // Plugins need to load using a class loader that includes the executor's user classpath
```

Comment:
Could you add comments to explain why we need to initialize plugin after heartbeater?